src/ethereum_test_vm/bytecode.py

```diff
@@ -148,12 +148,35 @@ def __add__(self, other: "Bytecode | int | None") -> "Bytecode":
         a_min, a_max = self.min_stack_height, self.max_stack_height
         b_pop, b_push = other.popped_stack_items, other.pushed_stack_items
         b_min, b_max = other.min_stack_height, other.max_stack_height
-        a_out = a_min - a_pop + a_push
 
-        c_pop = max(0, a_pop + (b_pop - a_push))
-        c_push = max(0, a_push + b_push - b_pop)
-        c_min = a_min if a_out >= b_min else (b_min - a_out) + a_min
-        c_max = max(a_max + max(0, b_min - a_out), b_max + max(0, a_out - b_min))
+        # NOTE: "_pop" is understood as the number of elements required by an instruction or
```

Comment:
I think this should definitely be split into its own PR, and also it would be great if we could add the test cases that this new approach covers and the previous implementation didn't (here: https://github.com/ethereum/execution-spec-tests/blob/061e41cce851e5d4d7de1d0612c84349f78b492e/src/ethereum_test_vm/tests/test_vm.py#L361).

Reply:
Just a tip: if you want to execute these framework tests specifically, you can simply run:
```bash
pytest -c pytest-framework.ini ./src/ethereum_test_vm/tests/test_vm.py::test_bytecode_properties
```

Reply:
Done: https://github.com/ethereum/execution-spec-tests/pull/810